pwiz_tools/Skyline/FileUI/MinimizeResultsDlg.cs

```diff
@@ -294,226 +243,183 @@ public void MinimizeToFile(string targetFile)
         {
             // First dispose of statistics collection to avoid it consuming disk reads
             // while actual minimizing is happening.
-            BlockStatisticsCollection = true;
+            _minimizeResults.PauseStatisticsCollection();
             if (!TryMinimizeToFile(targetFile))
             {
                 // Restore statistics calculation if minimizing didn't happen and the
                 // form is to remain open
-                BlockStatisticsCollection = false;
-                StatisticsCollector = new BackgroundWorker(this, null);
+                _minimizeResults.StartStatisticsCollection();
                 return;
             }
             DialogResult = DialogResult.OK;
         }
 
         public bool TryMinimizeToFile(string targetFile)
         {
-            var targetSkydFile = ChromatogramCache.FinalPathForName(targetFile, null);
-            using (var skydSaver = new FileSaver(targetSkydFile))
-            using (var scansSaver = new FileSaver(targetSkydFile + ChromatogramCache.SCANS_EXT, true))
-            using (var peaksSaver = new FileSaver(targetSkydFile + ChromatogramCache.PEAKS_EXT, true))
-            using (var scoreSaver = new FileSaver(targetSkydFile + ChromatogramCache.SCORES_EXT, true))
+            Exception exception = null;
+            using (var longWaitDlg = new LongWaitDlg(DocumentUIContainer))
             {
-                skydSaver.Stream = File.OpenWrite(skydSaver.SafeName);
-                using (var longWaitDlg = new LongWaitDlg(DocumentUIContainer))
-                {
-                    longWaitDlg.PerformWork(this, 1000,
-                        longWaitBroker =>
+                longWaitDlg.PerformWork(this, 1000,
+                    longWaitBroker =>
+                    {
+                        longWaitBroker.Message = Resources.MinimizeResultsDlg_MinimizeToFile_Saving_new_cache_file;
+                        try

```

Comment:
The threading change on exception handling makes me a little uneasy. Are you sure this is better? What is there now has obviously been in use for a while.